### ISSUE TITLE
Can you give more explanation on how to run in Mac M1 and Linux env

### ISSUE THREAD

**Issue by larrycai** [lorserker/ben#70]
>>===

For Mac M1/M2 users: you need to install [Homebrew](https://brew.sh/) if you don't have it already. Then, you need to brew install boost. You might need to open ./bin/darwin/libdds.so in XCode before launching ben for the first time.

For __Ubuntu__ users: You might have to compile libdds.so and might have to install libboost using `sudo apt install libboost-thread-dev` 

>>===

- I have python environment (create `requirements.txt` for all libraries)
- install `boost`
- saw the `bin/darwin/libdds.so`

Not so clear how to enable xcode (I don't have this toolkit) for it and why? (connect to python)

Now I start `python3 gameserver.py` under `src`, I got error

````
 File "git/github/ben/.venv/lib/python3.11/site-packages/pythonnet/__init__.py", line 82, in _create_runtime_from_spec
    raise RuntimeError(
RuntimeError: Failed to create a default .NET runtime, which would
                    have been "mono" on this system. Either install a
                    compatible runtime or configure it explicitly via
                    `set_runtime` or the `PYTHONNET_*` environment variables
                    (see set_runtime_from_env).
````

It is the same case for Linux env. (I would like to contribute for container image for linux env, so it is easily to be demoed)

**Comment by larrycai** [comment]
`src/dds.py` works for `darwin/libdds.so`, where can I find information for `EPBot86`

**Comment by larrycai** [comment]
related with #60 #36 (`dds` part), but they are not related with `gameserver.py` which need `EPBot` https://sites.google.com/view/bbaenglish/program-description

**Comment by larrycai** [comment]
Share some progress, install the default old mono framework, it is started

````
FROM ubuntu:jammy

RUN apt-get update && \
    apt-get -y install libboost-thread-dev python3.10 python-is-python3 python3-pip  && \
    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.10 1 && \
    apt-get clean

# https://linuxhint.com/install-use-mono-ubuntu-22-04/

#RUN echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" | tee /etc/apt/sources.list.d/mono-official-stable.list 
RUN apt-get update && \
    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends mono-complete 
ADD . /app

WORKDIR /app
RUN pip install -r requirements.txt
````

**Comment by ThorvaldAagaard** [comment]
There is no MAC-version of the EPBot86.

But interesting if we can run it in Mono.

The include of BBA should be move to where it is used, and the use of pythonnet can then be avoided on MAC, as BBA (EPBot86) is intended for testing only

But the docker files and installation are finem but perhaps you could include how you set start the docker images (or deploy it to a server)

**Comment by larrycai** [comment]
Oops, both works for Mac M1 (ARM) and Linux env using docker env ;-), at least it can be a simple solution who has knowledge of container. (see background https://opensource.com/article/22/5/guide-containers-images).

````
podman run -it -p 8080:8080 -p 4443:4443 ben
````

Then open browser to http://localhost:8080/home

It can be run in cloud environment easily (like kubernetes)

BTW: I thought Mac M1 needs EPBotARM86 from https://sites.google.com/view/bbaenglish/program-description, interesting EPBot86 works (need more knowledge to understand this)

**Comment by larrycai** [comment]
PR #71 is attached, recommend to merge it first to get more feedback, it will not impact current codes and setup on Windows, I can do further support on this area.

Be aware, the default mono in ubuntu:jammy is not latest, but it works so far, so just use it to make it clean in Dockerfile

**Comment by larrycai** [comment]
To make it more complete

* pure Mac env, it is solved by https://github.com/lorserker/ben/commit/6e2ffb372540d0237e77f5b81f69db7a88d8cdf2 form @ThorvaldAagaard , now we can run it natively on Mac
* for Mac/container env, libdds.so is not correct, I will fix it. needs arm based libdds.so.

**Comment by larrycai** [comment]
one more, in Dockerfile, for ubuntu we can use `libdds-dev` https://packages.ubuntu.com/search?keywords=libdds-dev , which includes `libdds.so`, it works for linux platform.

mac/linux arm version can be loaded, but get crash after 1-2 API calls in `game.py`

**Comment by ThorvaldAagaard** [comment]
What type of crash and where in game.py? I assume it is related to the platform and not game.py.

**Comment by larrycai** [comment]
I just got it is Killed (SIGKILL)

haha, so nice, after I recreate the container image, it works !! now I don't need `bin/*.so` folder for linux env on Mac/Linux.

**Comment by larrycai** [comment]
With PR #72 it is fixed (at least now), also see #13 for explanation for different env.
